omnibus/files/private-chef-cookbooks/private-chef/providers/pg_database.rb

```diff
@@ -10,33 +10,20 @@ def whyrun_supported?
 use_inline_resources
 
 action :create do
-  execute "create_database_#{new_resource.database}" do
-    command createdb_command
-    user node['private_chef']['postgresql']['username']
-    not_if {database_exist?}
-    retries 30
+  ruby_block "Maybe create database #{new_resource.database}" do
+    block do
+      EcPostgres.with_connection(node) do |connection|
+        result = connection.exec("SELECT datname FROM pg_database WHERE datname='#{new_resource.database}'")
+        if result.ntuples == 0
+          converge_by("Create database #{new_resource.database}") do
+              owner = "WITH OWNER #{new_resource.owner}" if new_resource.owner
+              connection.exec("CREATE DATABASE \"#{new_resource.database}\" #{owner} TEMPLATE #{new_resource.template} ENCODING '#{new_resource.encoding}';")
+              new_resource.updated_by_last_action(true)
+          end
+        else
+          new_resource.updated_by_last_action(false)
```

Comment:
Similarly, I think this can be left off.  That would reduce this section of code to:

```ruby
action :create do
  EcPostgres.with_connection(node) do |connection|
    result = connection.exec("SELECT datname FROM pg_database WHERE datname='#{new_resource.database}'")
    if result.ntuples == 0
      converge_by("Create database #{new_resource.database}") do
        owner = "WITH OWNER #{new_resource.owner}" if new_resource.owner
        connection.exec("CREATE DATABASE \"#{new_resource.database}\" #{owner} TEMPLATE #{new_resource.template} ENCODING '#{new_resource.encoding}';")
      end
    end
  end
end
```